language-plutus-core/test/Spec.hs

```diff
@@ -163,21 +163,19 @@ testsRewrite
     = testGroup "golden rewrite tests"
     . fmap (asGolden (format $ debugPrettyConfigPlcClassic defPrettyConfigPlcOptions))
 
-appAppLamLam :: MonadQuote m => m (Type TyNameWithKind ())
-appAppLamLam = do
-    x <- liftQuote (TyNameWithKind <$> freshTyName ((), Type ()) "x")
-    y <- liftQuote (TyNameWithKind <$> freshTyName ((), Type ()) "y")
-    pure $
-        TyApp ()
-            (TyApp ()
-                 (TyLam () x (Type ()) (TyLam () y (Type ()) $ TyVar () y))
-                 (TyBuiltin () TyInteger))
-            (TyBuiltin () TyInteger)
-
-testLam :: Either (TypeError ()) String
-testLam = fmap prettyPlcDefString . runQuote . runExceptT $ runTypeCheckM (TypeConfig True mempty Nothing) $ do
-    (TypeConfig _ _ gas) <- ask
-    normalizeType gas =<< appAppLamLam
+integer2 :: Type tyname ()
+integer2 = TyApp () (TyBuiltin () TyInteger) (TyInt () 2)
+
+getAppAppLamLam :: MonadQuote m => m (Type TyNameWithKind ())
+getAppAppLamLam = do
+    x <- liftQuote $ TyNameWithKind <$> freshTyName ((), Type ()) "x"
+    y <- liftQuote $ TyNameWithKind <$> freshTyName ((), Type ()) "y"
+    pure $ mkIterTyApp ()
+        (TyLam () x (Type ()) (TyLam () y (Type ()) $ TyVar () y))
+        [integer2, integer2]
```

Comment:
This was a strange test case previously: we checked that an ill-kinded type normalizes correctly to something well-kinded. Now we check that a well-kinded type (unchecked) normalizes correctly.